bin/cheribsdtest/cheribsdtest.c

```diff
@@ -622,6 +627,70 @@ cheribsdtest_run_test_name(const char *name)
 	errx(EX_USAGE, "unknown test: %s", name);
 }
 
+static void
+cheribsdtest_run_child(struct cheri_test *ctp)
+{
+	if (ctp->ct_child_func == NULL)
+		errx(EX_SOFTWARE, "%s has no child function", ctp->ct_name);
+	ctp->ct_child_func(ctp);
+	errx(EX_SOFTWARE, "%s child function returned", ctp->ct_name);
+}
+
+static void
+cheribsdtest_run_child_name(const char *name)
+{
+	struct cheri_test **ctpp, *ctp;
+
+	SET_FOREACH(ctpp, cheri_tests_set) {
+		ctp = *ctpp;
+		if (strcmp(name, ctp->ct_name) == 0) {
+			cheribsdtest_run_child(ctp);
+		}
+	}
+	errx(EX_USAGE, "unknown test: %s", name);
+}
+
+static char **
+mk_exec_args(const struct cheri_test *ctp)
+{
+	char *execpath;
+	char const **exec_args;
+	int argc = 0, error;
+
+	execpath = malloc(MAXPATHLEN);
+	if (execpath == NULL)
+		err(EX_OSERR, "malloc");
+	exec_args = calloc(5, sizeof(*exec_args));
+	if (exec_args == NULL)
+		err(EX_OSERR, "calloc");
+
+	/*
+	 * XXXBD: it would be nice if there was a way to say "coexecve
+	 * myself".
+	 */
+	error = elf_aux_info(AT_EXECPATH, execpath, MAXPATHLEN);
```

Comment:
FWIW this won't work with `/libexec/ld-elf.so.1 /bin/cheribsdtest-purecap`, but maybe that's ok

Reply:
Is there even a sensible option in the direct exec case?

Reply:
No… but a coexec_self(2) would fix that

Reply:
(Or 3 that rtld hooks I guess?)